cmd/state-exec/main.go

```diff
@@ -59,16 +59,28 @@ func run() error {
 	logr.Debug("hello")
 	defer logr.Debug("run goodbye")
 
-	hb, err := newHeartbeat()
+	a, err := newAttempt()
+	if err != nil {
+		return fmt.Errorf("cannot create new attempt: %w", err)
+	}
+	logr.Debug("message data - exec: %s", a.ExecPath)
+
+	hb, err := newHeartbeat(a.ExecPath)
```

Comment:
nit: I'd prefer to see the heartbeat construction closer to its usage below around line 91.